scripts/check-cryptohasher-symbols.py

```diff
@@ -0,0 +1,63 @@
+'''
+Today if 2 structs/enums use `CryptoHasher` derive and share the same name,
+the current `CryptoHasher` implementation does not prevent hash input collision.
+This can be a potential vulnerability.
+
+The easiest way is to let aptos developers ensure unique symbol names.
+
+This script is a quick and dirty script to help find enum/structs in this repo that
+use `CryptoHasher` derive and share the same name.
+'''
+
+from collections import defaultdict
+from pprint import pprint
+import os
+import re
+import subprocess
+
+# False positives that needs to be skipped for now.
+whitelisted_symbols = set([
+    'TestAptosCrypto',
+])
+
+ignored_crates = set([
+    'ahash',
+    'aptos_sdk',
+    'seahash',
+    'siphasher',
+])
+
+proc = subprocess.run("cargo doc --workspace --document-private-items", shell=True)
+assert proc.returncode == 0
+assert os.path.exists('target/doc')
+
+symbol_to_paths_map = defaultdict(set)
```

Comment:
it would be nice if this was unittested below or possible to test in some way

but this is small / unlikely to change much for now so go for it

https://github.com/aptos-labs/aptos-core/blob/main/testsuite/forge_test.py#L893-L908